### PR TITLE
feat: creating dateRangePicker

### PR DESCRIPTION
**Pull request by victorstassun**
### Description

Creating the new Date Range Picker component.

### Extra info

Passed design Q.A!!! 👍 

### Screenshots

Closed Input: 

image

Any date selected:

image

Only one date selected:

image

Range date selected:

image

### CODE REVIEW

File: src/components/Inputs/DateRangePicker/DateRangePicker.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      type: 'String',
+      defaultValue: 'right',
+      control: {
+        type: 'select',
+        options: ['right', 'left'],
+      },
+    },
+    inputStyle: {
+      type: 'String',
+      defaultValue: 'backgroundp-light',
```
Comment (gabrielforster) [inline src/components/Inputs/DateRangePicker/DateRangePicker.stories.mdx:23]:
```suggestion
      defaultValue: 'background-light',
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 2]
```diff
@@ -0,0 +1,913 @@
...
+                      @click.native="changeMonth('startDate', 'prev')"
+                    />
+                  </div>
+                  <div class="selector">
+                    <div
+                      :class="
+                        state.isMonthSelectorOpen.startDate
+                          ? 'input input-opened'
+                          : 'input'
+                      "
```
Comment (gabrielforster) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:57]: computed for a cleaner code

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 3]
```diff
@@ -0,0 +1,913 @@
...
+                    />
+                  </div>
+                </div>
+                <div class="selector">
+                  <div
+                    :class="
+                      state.isYearSelectorOpen.startDate
+                        ? 'input input-opened'
+                        : 'input'
+                    "
```
Comment (gabrielforster) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:97]: computed?

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 4]
```diff
@@ -0,0 +1,913 @@
...
+                      @click.native="changeMonth('endDate', 'prev')"
+                    />
+                  </div>
+                  <div class="selector">
+                    <div
+                      :class="
+                        state.isMonthSelectorOpen.endDate
+                          ? 'input input-opened'
+                          : 'input'
+                      "
```
Comment (gabrielforster) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:165]: computed?

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 5]
```diff
@@ -0,0 +1,913 @@
...
+                    />
+                  </div>
+                </div>
+                <div class="selector">
+                  <div
+                    :class="
+                      state.isYearSelectorOpen.endDate
+                        ? 'input input-opened'
+                        : 'input'
+                    "
```
Comment (gabrielforster) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:205]: computed?

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 6]
```diff
@@ -0,0 +1,913 @@
...
+    days(dateType, monthOffset = 0) {
+      const date = new Date(this.state.calendaryDates[dateType]);
+      date.setMonth(date.getMonth() + monthOffset);
+
+      const month = date.getMonth();
+      const year = date.getFullYear();
+
+      const firstDay = new Date(year, month, 1).getDay();
+      const daysInMonth = new Date(year, month + 1, 0).getDate();
+      const days = [...Array(daysInMonth).keys()].map(i => i + 1);
```
Comment (gabrielforster) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:478]:
```suggestion
      const days = [...Array(daysInMonth).keys()].map(day => day + 1);
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 7]
```diff
@@ -0,0 +1,913 @@
...
+            startDate,
+            endDate: newEndDate,
+          };
+        }
+      }
+    },
+
+    daysOnRange(day) {
+      const { startDate } = this.state.inputValue;
+      const { endDate } = this.state.inputValue;
```
Comment (gabrielforster) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:532]:
```suggestion
      const { startDate, endDate } = this.state.inputValue;
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 8]
```diff
@@ -0,0 +1,913 @@
...
+          endDate: new Date(),
+        },
+        'Últimos 6 meses': {
+          startDate: new Date(new Date().getTime() - 15552000000),
+          endDate: new Date(),
+        },
+        'Últimos 12 meses': {
+          startDate: new Date(new Date().getTime() - 31104000000),
+          endDate: new Date(),
+        },
```
Comment (gabrielforster) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:559]: Wouldn't be cool to use `decreaseDate` from adapcon-utils-js?
Is a simple function that receives a date and a certain duration do be decreased. I think its better than having numbers like this on the code

**Review by gabrielforster — CHANGES_REQUESTED** [review]
Small changes for better code!

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue (hunk 8 above)
Comment (victorstassun) [reply src/components/Inputs/DateRangePicker/DateRangePicker.vue:559]: Of course, I didn´t know about it, but I will change it.

**Review by caroltrindade — CHANGES_REQUESTED** [review]

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 9]
```diff
@@ -0,0 +1,903 @@
...
+        'default',
+        {
+          month: 'long',
+        },
+      );
+
+      const abreviatedMonth = month.slice(0, 3);
+
+      const capitalizedMonth = abreviatedMonth.slice(0, 1).toUpperCase() + abreviatedMonth.slice(1);
+      return capitalizedMonth;
```
Comment (gabrielforster) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:399]:
```suggestion
      return abreviatedMonth.slice(0, 1).toUpperCase() + abreviatedMonth.slice(1);
```

**Review by gabrielforster — APPROVED** [review]
LGTM

**Review by gabrielforster — DISMISSED** [review]
LGTM

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 10]
```diff
@@ -0,0 +1,904 @@
...
+        } else {
+          this.state.inputValue = {
+            startDate,
+            endDate: newEndDate,
+          };
+        }
+      }
+    },
+
+    daysOnRange(day) {
```
Comment (caroltrindade) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:520]: image
The date format shouldn't be DD/MM/YYYY? Like 03/04/2023

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 11]
```diff
@@ -0,0 +1,904 @@
+<template>
+  <section>
+    <div class="input-container">
+      <input
+        v-model="inputValue"
+        class="pb"
+        :style="getInputStyle"
+        :placeholder="inputValue"
+        type="text"
+        @click="showPicker"
```
Comment (caroltrindade) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:10]:
```suggestion
        @click="toggleDatePicker"
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 12]
```diff
@@ -0,0 +1,904 @@
...
+                  @click="selectDay('endDate', day.id)"
+                >
+                  {{ day.id }}
+                </p>
+              </div>
+            </div>
+          </div>
+          <div class="buttons-container">
+            <div class="period-visualizer">
+              <p class="pb-md">{{ showSelectedPeriod() }}</p>
```
Comment (caroltrindade) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:239]:
```suggestion
              <p class="pb-md">{{ selectedPeriod() }}</p>
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 13]
```diff
@@ -0,0 +1,904 @@
...
+                      button-size="large"
+                      icon-position="left"
+                      icon="fas fa-chevron-up"
+                      @click.native="changeMonth('startDate', 'prev')"
+                    />
+                  </div>
+                  <div class="selector">
+                    <div
+                      :class="getSelectorClass('month')"
+                      @click="openMonthsSelector('startDate')"
```
Comment (caroltrindade) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:54]:
```suggestion
                      @click="toggleMonthsSelector('startDate')"
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 14]
```diff
@@ -0,0 +1,904 @@
...
+                      icon-position="left"
+                      icon="fas fa-chevron-up"
+                      @click.native="changeMonth('startDate', 'next')"
+                    />
+                  </div>
+                </div>
+                <div class="selector">
+                  <div
+                    :class="getSelectorClass('year')"
+                    @click="openYearsSelector('startDate')"
```
Comment (caroltrindade) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:90]:
```suggestion
                    @click="toggleYearsSelector('startDate')"
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 15]
```diff
@@ -0,0 +1,904 @@
...
+                label="Redefinir"
+                button-style="background-light"
+                @click.native="resetDates()"
+              />
+              <PbButton
+                style="width: 86px"
+                color="primary"
+                label="Aplicar"
+                button-style="background"
+                @click.native="applyFilter()"
```
Comment (caroltrindade) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:254]:
```suggestion
                @click.native="apply()"
```
or
```suggestion
                @click.native="saveDates()"
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 16]
```diff
@@ -0,0 +1,904 @@
...
+      const abreviatedMonth = month.slice(0, 3);
+
+      return abreviatedMonth.slice(0, 1).toUpperCase() + abreviatedMonth.slice(1);
+    },
+
+    getSelectorClass(selectorType) {
+      if (selectorType === 'year')
+        return this.state.isYearSelectorOpen.startDate ? 'input input-opened' : 'input';
+      return this.state.isMonthSelectorOpen.startDate ? 'input input-opened' : 'input';
+    },
```
Comment (caroltrindade) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:405]:
```suggestion
    getSelectorClass(selectorType) {
      const bySelector = { month: this.state.isMonthSelectorOpen, year: this.state.isYearSelectorOpen }
      return bySelector[selectorType].startDate ? 'input input-opened' : 'input';
    },
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 17]
```diff
@@ -0,0 +1,904 @@
...
+          'Ago',
+          'Set',
+          'Out',
+          'Nov',
+          'Dez',
+        ],
+        isMonthSelectorOpen: {
+          startDate: false,
+          endDate: false,
+        },
```
Comment (caroltrindade) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:307]:
```suggestion
        monthSelector: {
          open: false,
          startDate: '',
          endDate: '',
        },
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 18]
```diff
@@ -0,0 +1,904 @@
...
+        yearOptions: [],
+        isYearSelectorOpen: {
+          startDate: false,
+          endDate: false,
+        },
```
Comment (caroltrindade) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:312]:
```suggestion
        isYearSelectorOpen: {
          open: false,
          startDate: '',
          endDate: '',
        },
```

**Review by caroltrindade — CHANGES_REQUESTED** [review]

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue (hunk 10 above)
Comment (gabrielforster) [reply src/components/Inputs/DateRangePicker/DateRangePicker.vue:521]: Changing it so the date is formatted via toLocaleDateString.

**Review by CauaKath — APPROVED** [review]
LGTM

**Review by gabrielforster — APPROVED** [review]
LGTM

**Review by mickaelnn — DISMISSED** [review]
LGTM

**Review by gabrielforster — COMMENTED** [review]
Some comments and suggestions.
This comments does not block the pr if it got 2 LGTM

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 19]
```diff
@@ -0,0 +1,908 @@
...
+        } else { this.state.inputValue.endDate = newEndDate };
+      }
+    },
+
+    daysOnRange(day) {
+      const { startDate, endDate } = this.state.inputValue;
+
+      const yesterday = startDate
+        ? new Date(startDate.getTime() - 86400000)
+        : new Date();
```
[the remark this answers is not in the document]
Comment (victorstassun) [reply src/components/Inputs/DateRangePicker/DateRangePicker.vue:585]: I think the comment might be helpful, but I don´t see any reasons to make all this calc.

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 20]
```diff
@@ -0,0 +1,908 @@
...
+            : 'var(--color-gray-40)',
+          border: this.state.isPickerVisible
+            ? '1px solid var(--color-gray-90)'
+            : '1px solid var(--color-gray-5) !important',
+          borderRadius: this.state.isPickerVisible ? '20px 20px 0 0' : '20px',
+        },
+      };
+
+      return inputStyles[this.inputStyle];
+    },
```
[the remark this answers is not in the document]
Comment (victorstassun) [reply src/components/Inputs/DateRangePicker/DateRangePicker.vue:372]: Did it, just changed the return of this function and removed the inputStyle object, now we return the hole style, but with a validation on the background-color

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue [hunk 21]
```diff
@@ -0,0 +1,908 @@
...
+        this.state.calendarsVisualization.startDate = newDate;
+      else this.state.calendarsVisualization.endDate = newDate;
+    },
+
+    selectMonth(dateType, month) {
+      const date = this.getCurrentDate(dateType);
+      const monthIndex = this.state.monthOptions.indexOf(month);
+      date.setMonth(monthIndex);
+
+
```
[the remark this answers is not in the document]
Comment (victorstassun) [reply src/components/Inputs/DateRangePicker/DateRangePicker.vue]: Already removed that blank space

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue (hunk 19 above)
[the remark this answers is not in the document]
Comment (gabrielforster) [reply src/components/Inputs/DateRangePicker/DateRangePicker.vue:569]: thats ok

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue (hunk 20 above)
[the remark this answers is not in the document]
Comment (gabrielforster) [reply src/components/Inputs/DateRangePicker/DateRangePicker.vue]: 🚀

**Review by gabrielforster — APPROVED** [review]
LGTM

**Review by mickaelnn — APPROVED** [review]
LGTM

**Pull request merged**